tests/integration/test_charm.py

```diff
@@ -124,134 +126,198 @@ async def test_dashboard_access_https(ops_test: OpsTest):
+    assert all(
+        unit.workload_status == "active" for unit in ops_test.model.applications[APP_NAME].units
     )
 
-    expected_results = [
-        {
-            "metrics_path": "/metrics",
-            "scheme": "http",
-        }
-    ]
-    agent_unit = ops_test.model.applications[COS_AGENT_APP_NAME].units[0]
-    for unit in ops_test.model.applications[APP_NAME].units:
-        unit_ip = await get_address(ops_test, unit.name)
-        relation_data = get_unit_relation_data(
-            ops_test.model.name, agent_unit.name, COS_AGENT_RELATION_NAME
-        )
-        expected_results[0]["static_configs"] = [{"targets": [f"{unit_ip}:9684"]}]
-        unit_data = relation_data[unit.name]
-        unit_cos_config = json.loads(unit_data["data"]["config"])
-        for key, value in expected_results[0].items():
-            assert unit_cos_config["metrics_scrape_jobs"][0][key] == value
-
-
-@pytest.mark.group(1)
-@pytest.mark.abort_on_fail
-@pytest.mark.log_level_change
-async def test_log_level_change(ops_test: OpsTest):
-
-    for unit in ops_test.model.applications[APP_NAME].units:
-        assert count_lines_with(
-            ops_test.model_full_name,
-            unit.name,
-            "/var/snap/opensearch-dashboards/common/var/log/opensearch-dashboards/opensearch_dashboards.log",
-            "debug",
-        )
-
-        await ops_test.model.applications[APP_NAME].set_config({"log_level": "ERROR"})
+    logger.info("Checking if Dashboards is available again")
+    opensearch_relation = get_relations(ops_test, OPENSEARCH_RELATION_NAME)[0]
+    assert access_all_dashboards(ops_test, opensearch_relation, https=True)
 
-        await ops_test.model.wait_for_idle(
-            apps=[APP_NAME], status="active", timeout=1000, idle_period=30
-        )
+    logger.info("Removing an opensearch unit so Opensearch gets in a 'red' state")
+    await ops_test.model.applications[APP_NAME].destroy_unit(
+        ops_test.model.applications[OPENSEARCH_APP_NAME].units[0].name
+    )
+    async with ops_test.fast_forward("30s"):
+        await ops_test.model.wait_for_idle(apps=[APP_NAME], status="blocked")
 
-        debug_lines = count_lines_with(
-            ops_test.model_full_name,
-            unit.name,
-            "/var/snap/opensearch-dashboards/common/var/log/opensearch-dashboards/opensearch_dashboards.log",
-            "debug",
-        )
+    assert await check_full_status(
+        ops_test, status="blocked", status_msg="Opensearch service is (partially or fully) down"
+    )
 
-        assert (
-            count_lines_with(
-                ops_test.model_full_name,
-                unit.name,
-                "/var/snap/opensearch-dashboards/common/var/log/opensearch-dashboards/opensearch_dashboards.log",
-                "debug",
-            )
-            == debug_lines
-        )
+    logger.info("Destroying and restoring the Opensearch cluster")
+    await destroy_cluster(ops_test, app=OPENSEARCH_APP_NAME)
```

Comment:
Unfortunately this doesn't seem stable.
https://github.com/canonical/opensearch-dashboards-operator/actions/runs/10175100391/job/28142472214#step:28:326

So I'm thinking to cut this out and add back a scaleup once the node election is fixed.

Reply:
@Mehdi-Bendriss Pls let me know your thoughts on this. I don't have a better idea to get green pipelines (with at least partial test coverage)

Reply:
Please comment it for now and create a new ticket, and add a TODO comment to un-comment once the "ticket id" is resolved.